### ISSUE TITLE
Using RegEx feature of finder xpaths leads to error

### ISSUE THREAD

**Issue by JPrevost** [MITLibraries/scoap3hub#54]
With this finder idKey: `//email @(.*)$` I get the follow error.

I haven't confirmed if it works in some cases or not.

This record is useful for that Finder.
http://repo.scoap3.org/record/4252/files/main.xml

```
[ERROR] [02/26/2015 13:48:19.064] [application-akka.actor.default-dispatcher-5] [akka://application/user/cataloger] jorge.casalderrey@ub.edu (of class java.lang.String)
scala.MatchError: jorge.casalderrey@ub.edu (of class java.lang.String)
	at workers.Cataloger$$anonfun$1.apply(Cataloger.scala:110)
	at workers.Cataloger$$anonfun$1.apply(Cataloger.scala:106)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:245)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:245)
	at scala.collection.Iterator$class.foreach(Iterator.scala:743)
	at scales.utils.collection.DuplicateFilter$$anon$1.foreach(DuplicateFilter.scala:20)
	at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
	at scales.utils.collection.DuplicateFilter.foreach(DuplicateFilter.scala:14)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:245)
	at scales.utils.collection.DuplicateFilter.map(DuplicateFilter.scala:14)
	at workers.Cataloger.workers$Cataloger$$process(Cataloger.scala:106)
	at workers.Cataloger$$anonfun$topics$1.apply(Cataloger.scala:72)
	at workers.Cataloger$$anonfun$topics$1.apply(Cataloger.scala:70)
	at scala.collection.immutable.List.map(List.scala:272)
	at workers.Cataloger.topics(Cataloger.scala:70)
	at workers.Cataloger$$anonfun$catalog$2.apply(Cataloger.scala:304)
	at workers.Cataloger$$anonfun$catalog$2.apply(Cataloger.scala:304)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at workers.Cataloger$.catalog(Cataloger.scala:304)
	at workers.CatalogWorker$$anonfun$receive$1.applyOrElse(Cataloger.scala:33)
	at akka.actor.Actor$class.aroundReceive(Actor.scala:465)
	at workers.CatalogWorker.aroundReceive(Cataloger.scala:31)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:516)
	at akka.actor.ActorCell.invoke(ActorCell.scala:487)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:238)
	at akka.dispatch.Mailbox.run(Mailbox.scala:220)
	at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:393)
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```

**Comment by richardrodgers** [comment]
Is the error reproducible if one substitutes the following:

  //email .*@(.*)$

?

**Comment by richardrodgers** [comment]
Sorry - the markup/down is wrong . Let me try again

`//email .*@(.*)$`

**Comment by JPrevost** [comment]
That still breaks (new error though).

It may be more efficient if you could replicate locally... if you use https://github.mit.edu/mitlibraries/scoap3_configuration/raw/master/cmodel.json and https://github.mit.edu/mitlibraries/scoap3_configuration/raw/master/pubmodel.json and then knip this you should see the problem: http://localhost:9000/knip/1/1/oai:repo.scoap3.org:9308?force=true

I think :)

**Comment by JPrevost** [comment]
Thanks for looking at this @richardrodgers 

This was all essentially an error on my part / confusion based on the label.

I always put the same xpath for the `id_key` and `id_label` since something needs to go there. The regex works for the `id_key` as expected but fails for the `id_label`. That's probably fine but we should probably consider whether we will use the label and remove it if not and update it to allow it to be blank if we keep it rather than having to populate it when it's not needed.

**Comment by richardrodgers** [comment]
We need to review the label thing, agreed. I have not briefed you on it's intended behavior, and there are dependencies on the XPath/XQuery direction we take. This is on my list.....
